channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -229,11 +237,17 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
                         _InternalMessageId.InvalidInstrumentationKey, "Invalid Instrumentation key "+config.instrumentationKey);
                 }
 
+                if (!isInternalApplicationInsightsEndpoint(_self._senderConfig.endpointUrl()) && _self._senderConfig.customHeaders() && _self._senderConfig.customHeaders().length > 0) {
+                    arrForEach(_self._senderConfig.customHeaders(), customHeader => {
+                        this.addHeader(customHeader.header, customHeader.value);
+                    });
+                }
+
                 if (!_self._senderConfig.isBeaconApiDisabled() && isBeaconApiSupported()) {
                     _self._sender = _beaconSender;
                 } else {
-                    if (typeof XMLHttpRequest !== undefined) {
-                        const xhr:any = getGlobalInst("XMLHttpRequest");
+                    if (typeof XMLHttpRequest !== "undefined") {
```

Comment:
Found a bug here - need to check type `"undefined"` instead of `undefined`. Otherwise this if check is always true.

Reply:
getGlobalInst("XMLHttpRequest") should return either null or undefined (don't remember) if it's not defined -- so you shouldn't need the typeof check...